apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,418 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
+    RetryThenFailHelper(service => service.Objects.Delete("bucket", "objectName"), client => client.DeleteObject("bucket", "objectName"));
+        */
+
+        [Fact]
+        public void UpdateObject_WithMetageneration_RetryOnce() =>
+           RetryOnceHelper(
+               service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+               client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void UpdateObject_WithoutMetageneration_NoRetry() =>
+            NoRetryHelper(
+                service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object"),
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions()));
+
+        [Fact]
+        public void UpdateObject_WithoutOptions_NoRetry() =>
+            NoRetryHelper(
+                service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object"),
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }));
+
+        [Fact]
+        public void UpdateObject_RetryThenFail() =>
+        RetryThenFailHelper(service => new ObjectsResource.UpdateRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+                client => client.UpdateObject(new Object { Bucket = "bucket", Name = "object" }, new UpdateObjectOptions { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void PatchObject_WithMetageneration_RetryOnce() =>
+          RetryOnceHelper(
+              service => new ObjectsResource.PatchRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+              client => client.PatchObject(new Object { Bucket = "bucket", Name = "object" }, new PatchObjectOptions { IfMetagenerationMatch = 70 }));
+
+        [Fact]
+        public void PatchObject_WithoutMetageneration_NoRetry() =>
+           NoRetryHelper(
+               service => new ObjectsResource.PatchRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object"),
+               client => client.PatchObject(new Object { Bucket = "bucket", Name = "object" }, new PatchObjectOptions()));
+
+        [Fact]
+        public void PatchObject_RetryThenFail() =>
+RetryThenFailHelper(service => new ObjectsResource.PatchRequest(service, new Object { Bucket = "bucket", Name = "object" }, "bucket", "object") { IfMetagenerationMatch = 70 },
+        client => client.PatchObject(new Object { Bucket = "bucket", Name = "object" }, new PatchObjectOptions { IfMetagenerationMatch = 70 }));
+
+        /*
```

Comment:
The tests you've got should be removed anyway. They're not testing the behaviour of `StorageClient`, which is what we're really trying to test.

The only StorageClient operation we've got that uses the Rewrite API is Copy, which you've got tests for above.